### DISCUSSION TITLE
[crud-service] Is there the possibility to return oas v3 instead of swagger v2 documentation?

### DISCUSSION THREAD

**Discussion by frengooo** [mia-platform/community#266] [Q&A] [answered]
### Which section?

Marketplace

### Which is the version?

v11

### Description

Hi guys,
I don't know if this is the right section, tell me if I need to move the question on the crud-service repository.

I need to know if I can "tell" the `crud-service` to return the documentation in `oas v3` version.
At the moment the crud returns a `swagger v2` documentation on the `/documentation/json` route.

I noticed that there is the following line [in the crud-service repository](https://github.com/mia-platform/crud-service/blob/4265714d5cc57f770eaf51117157c00ec68253c0/index.js#L263):
<img width="346" alt="Screenshot 2023-06-08 at 11 32 40" src="https://github.com/mia-platform/community/assets/28152586/c3d7e1ee-f2af-4fba-9a56-589bd67ff948">

By commenting or removing that line the returned documentation is in `oas v3`.

Thanks in advance.

**Comment by davidebianchi** [comment, marked as answer]
As I know, patternProperties are not supported in oas v3.0 (it will be supported by v3.1), but crud-service uses patternProperties in the schema, which in the past result in an error to generate it in oas3. Maybe it could be retried.

**Reply by frengooo** [reply]
You are right. The generated oas is not valid 😢 .

**Reply by danibix95** [reply]
@silversoul93 Could you please open an issue on the CRUD Service to keep track of this?

**Reply by frengooo** [reply]
@danibix95 Done https://github.com/mia-platform/crud-service/issues/100
